mds/authent/views.py

```diff
@@ -0,0 +1,65 @@
+import base64
+import binascii
+import json
+from urllib.parse import unquote_plus
+
+from django.http import HttpResponse
+from django.utils.crypto import constant_time_compare
+from django.utils.decorators import method_decorator
+from django.views.generic import View
+from django.views.decorators.csrf import csrf_exempt
+
+from . import generators
+from . import models
+from . import settings
+
+
+@method_decorator(csrf_exempt, name="dispatch")
+class LongLivedTokenView(View):
+    """
+    Implements an endpoint to provide long lived (JWT) tokens.
+
+    This is an alternate authentication method to Oauth2.
+    """
+
+    def post(self, request, *args, **kwargs):
+        application = self.authenticate_client(request)
+        if not application:
+            response = HttpResponse(status=401)
+            response["WWW-Authenticate"] = b"Basic"
+            return response
+        token_duration = settings.AUTHENT_LONG_LIVED_TOKEN_DURATION
+        token = generators.generate_jwt(application)
+        data = {"access_token": "%s", "token_type": "bearer"}
+        if token_duration:
+            data["expires_in"] = int(token_duration.total_seconds())
+        payload = json.dumps(data).encode("utf-8") % token
+        return HttpResponse(
+            payload, content_type="application/json; charset=utf-8", status=200
+        )
+
+    def authenticate_client(self, request):
```

Comment:
Why use another authentication mean ?
Long-lived token generation will be done through the backoffice that will use normal JWT. We could simple check that the RemoteUser has the required scope and rights on provider